web/modules/custom/joinup_core/src/JoinupWorkflowAccessControlHandler.php

```diff
@@ -0,0 +1,279 @@
+   * Access check for delete operation.
+   *
+   * Delete operation should not be interrupted by states. The user either has
+   * or has not the permission.
+   * The following checks take place with the given priority:
+   * - If the user has global permission to delete any entity of the given
+   * bundle, he is being granted access.
+   * - If the user has group permission to delete any entity of the given
+   * bundle, he is being granted access.
+   * - If the user is the owner of the entity, he is allowed only if the parent
+   * is in a post moderated state.
+   * In all other cases, we allow entity access handler to decide.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *    The entity object.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *    The user account.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *    The access result.
+   */
+  protected function entityDeleteAccess(EntityInterface $entity, AccountInterface $account) {
+    $entity_type = ($entity->getEntityTypeId() == 'node') ? 'content' : 'rdf_entity';
+    if ($account->hasPermission("delete any {$entity->bundle()} {$entity_type}")) {
+      return AccessResult::allowed();
+    }
+
+    $parent = $this->getEntityParent($entity);
+    $membership = Og::getMembership($parent, $account);
+    if (!empty($membership) && $membership->hasPermission("delete any {$entity->bundle()} {$entity_type}")) {
+      return AccessResult::allowed();
+    }
+
+    $workflow = $this->getEntityWorkflow($entity);
+    if ($workflow === self::WORKFLOW_PRE_MODERATED) {
+      return AccessResult::forbiddenIf(!$account->hasPermission("delete any {$entity->bundle()} {$entity_type}"));
+    }
+
+    return AccessResult::neutral();
+  }
+
+  /**
+   * Helper method to retrieve the parent of the entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The group content entity.
+   *
+   * @return \Drupal\rdf_entity\RdfInterface|null
+   *   The rdf entity the entity belongs to, or NULL when no group is found.
+   */
+  protected function getEntityParent(EntityInterface $entity) {
+    $groups = $this->membershipManager->getGroups($entity);
+
+    if (empty($groups['rdf_entity'])) {
+      return NULL;
+    }
+
+    return reset($groups['rdf_entity']);
+  }
+
+  /**
+   * Returns the appropriate workflow to use for the passed entity.
+   *
+   * If the entity is a group, then the workflow is the default workflow.
+   * If the entity is a group content, then the workflow is dependant to the
+   * moderation settings of the parent.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *    The group content entity.
+   *
+   * @return string
+   *    The id of the workflow to use.
+   */
+  protected function getEntityWorkflow(EntityInterface $entity) {
+    if (Og::isGroup($entity->getEntityTypeId(), $entity->bundle())) {
+      return self::WORKFLOW_DEFAULT;
+    }
+
+    $parent = $this->getEntityParent($entity);
```

Comment:
This has already been removed in the backport and the RelationManager is being injected to the NodeGuard.